test/cmd/image-lookup.sh

```diff
@@ -13,45 +13,48 @@ trap os::test::junit::reconcile_output EXIT
 project="$( oc project -q )"
 
 os::test::junit::declare_suite_start "cmd/image-lookup"
-## This test validates image lookup resolution
-#
-## TODO: fix and re-enable these tests before 4.0 release
-#
-## Verify image resolution on default resource types
-#os::cmd::expect_success_and_text "oc import-image --confirm --from=nginx:latest nginx:latest" "sha256:"
-#os::cmd::expect_success_and_text "oc set image-lookup is/nginx" "updated"
-## Image lookup works for pods
-#os::cmd::expect_success          "oc run --generator=run-pod/v1 --restart=Never --image=nginx:latest nginx"
+# This test validates image lookup resolution
+
+# Verify image resolution on default resource types
+os::cmd::expect_success_and_text "oc import-image --confirm --from=nginx:latest nginx:latest" "sha256:"
+os::cmd::expect_success_and_text "oc set image-lookup is/nginx" "updated"
+# Image lookup works for pods
+os::cmd::expect_success          "oc run --generator=run-pod/v1 --restart=Never --image=nginx:latest nginx"
+# TODO: the container image resolves to "nginx:latest", not the actual image ref. Why?
 #os::cmd::expect_success_and_text "oc get pod/nginx -o jsonpath='{.spec.containers[0].image}'" "nginx@sha256:"
-## Image lookup works for jobs
-#os::cmd::expect_success          "oc run --generator=job/v1 --restart=Never --image=nginx:latest nginx"
```

Comment:
i'm sure this is a result of the imageadmissionpolicy plugin not being configured correctly on the kubeapi server currently.

hopefully @deads2k understands the state of that configuration/admission plugin enablement better than I do.

Reply:
@deads2k to clarify the specific issue here: the container `image:` field for pods is getting set to `imageName:latest` instead of the image sha256 value

Reply:
@bparees from local testing, I'm seeing that the image admission gets stuck [here](https://github.com/openshift/origin/blob/master/pkg/image/apiserver/admission/imagepolicy/imagepolicy.go#L43) due to a nil `input` reader

Reply:
this is, hopefully, the PR you are waiting for: https://github.com/openshift/origin/pull/21458

Reply:
@bparees Thanks, although I tried it locally, and since we now instantiate an imagepolicy plugin with an empty config by default, that means that it pretty much will not "accept" any resources :/

Reply:
these are the rules i'd expect to see getting setup by default: https://github.com/openshift/origin/blob/master/pkg/image/apiserver/admission/apis/imagepolicy/v1/defaults.go#L8

Reply:
(which would mean no execution rules, thus nothing should be blocked)

Reply:
There are resolution rules, they get defaulted:
https://github.com/openshift/origin/blob/master/pkg/image/apiserver/admission/apis/imagepolicy/v1/defaults.go#L23-L34

they should all get defaulted to DoNotAttempt:
https://github.com/openshift/origin/blob/master/pkg/image/apiserver/admission/apis/imagepolicy/v1/defaults.go#L40

And there are no execution rules.  If having no execution rules means resources are rejected i'll need to you to point me to where, I don't believe that's the behavior but i don't have time to go through the admission controller right now.

That said I do not think we actually want to resolutionrule to be DoNotAttempt.  We probably need to create a default execution rule or otherwise change the resolutionrule defaulting behavior.

(note that ExecutionRule and ResolutionRule are two distinct things and they are only related by this line: https://github.com/openshift/origin/blob/master/pkg/image/apiserver/admission/apis/imagepolicy/v1/defaults.go#L42 i believe, but again I don't have time right now to go rediscover how this thing works)

Reply:
@bparees 
> And there are no execution rules. If having no execution rules means resources are rejected i'll need to you to point me to where, I don't believe that's the behavior but i don't have time to go through the admission controller right now.

you're right, there are no execution rules. No, I think resources were being rejected due to no resolution rules. It looks like the config was just not being defaulted. I manually default resolution rules In the second-to-last commit (only to see if this would solve the problem), and that does indeed restore the desired behavior for the remaining `image-lookup` tests.

> That said I do not think we actually want to resolutionrule to be DoNotAttempt. We probably need to create a default execution rule or otherwise change the resolutionrule defaulting behavior.

I am not familiar enough with how this works, but can attempt it in a separate PR.

It looks like defaulting is done on `v1` policyConfig objects, but we are using an internal policyConfig object here, which is my guess as to why it is not being defaulted with the expected resolution rules.